lib/plugins/wallet/tron/base.js

```diff
@@ -0,0 +1,190 @@
+  const transactionInfo = tronWeb.trx.getTransactionInfo(txId)
+
+  if (!transactionInfo || !transactionInfo.fee) return { txId }
+
+  const fee = new BN(transactionInfo.fee).decimalPlaces(0)
+  return { txId, fee }
+}
+
+const generateTrc20Tx = async (toAddress, wallet, amount, cryptoCode) => {
+  const contractAddress = coins.utils.getTrc20Token(cryptoCode).contractAddress
+  const functionSelector = 'transfer(address,uint256)'
+  const parameters = [
+    { type: 'address', value: tronWeb.address.toHex(toAddress) },
+    { type: 'uint256', value: amount }
+  ]
+
+  const tx = await tronWeb.transactionBuilder.triggerSmartContract(contractAddress, functionSelector, {}, parameters, wallet.address)
+
+  return tronWeb.trx.sign(tx.transaction, wallet.privateKey.slice(2))
+}
+
+const generateTx = async (toAddress, wallet, amount) => {
+  const transaction = await tronWeb.transactionBuilder.sendTrx(toAddress, amount, wallet.address)
+
+  const privateKey = wallet.privateKey
+
+  // their api return a hex string starting with 0x but expects without it
+  return tronWeb.trx.sign(transaction, privateKey.slice(2))
+}
+
+function newFunding (account, cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(code => {
+      const fundingAddress = defaultAddress(account)
+
+      return confirmedBalance(fundingAddress, code)
+        .then((balance) => ({
+          fundingPendingBalance: BN(0),
+          fundingConfirmedBalance: balance,
+          fundingAddress
+        }))
+    })
+}
+
+function sweep (account, txId, cryptoCode, hdIndex) {
+  const wallet = paymentWallet(account, hdIndex)
+  const fromAddress = wallet.address
+  const isTrc20Token = coins.utils.isTrc20Token(cryptoCode)
+
+  const txFunction = isTrc20Token ? generateTrc20Tx : generateTx
+
+  return SWEEP_QUEUE.add(async () => {
+    const r = await confirmedBalance(fromAddress, cryptoCode)
+    if (r.eq(0)) return
+    const signedTx = await txFunction(defaultAddress(account), wallet, r.toString(), cryptoCode)
+    let response = null
+    try {
+      response = await tronWeb.trx.sendRawTransaction(signedTx)
+      if (!response.result) throw new Error(response.code)
+    } catch (err) {
+      // for some reason err here is just a string
+      throw new Error(err)
+    }
+    return response
+  })
+}
+
+function connect(account) {
+  if (tronWeb != null) return
+  const endpoint = account.endpoint
+  const apiKey = account.apiKey
+  tronWeb = new TronWeb({
+    fullHost: endpoint,
+    headers: { "TRON-PRO-API-KEY": apiKey },
+    privateKey: '01'
+  })
+}
+
+function getStatus (account, tx, requested, settings, operatorId) {
+  const { toAddress, cryptoCode } = tx
+  return checkCryptoCode(cryptoCode)
+    .then(code => confirmedBalance(toAddress, code))
+    .then((confirmed) => {
+      if (confirmed.gte(requested)) return { receivedCryptoAtoms: confirmed, status: 'confirmed' }
+      return { receivedCryptoAtoms: 0, status: 'notSeen' }
+    })
```

Comment:
Isn't it possible to get a transaction of part of the requested value? Like with the other wallets:

https://github.com/lamassu/lamassu-server/blob/e3892e9ac2f111fedf26545441c18419250cb773/lib/plugins/wallet/bitcoind/bitcoind.js#L148

Reply:
Good catch, updated.

Reply:
Actually the problem here is a different one, we don't really have pending amount. Same thing happens with ETH so I'll get confirmation from tech first to see what to do here.